pkg/installer/custominstallconfig_test.go

```diff
@@ -363,3 +374,54 @@ func verifyIgnitionFiles(t *testing.T, temp map[string]any, storageFiles []strin
 	}
 	assert.EqualValues(t, "ARO", config.Data["invoker"])
 }
+
+func verifyMasterPointerIgnition(t *testing.T, ignData []byte) {
+	ignContents := &igntypes.Config{}
+	err := json.Unmarshal(ignData, &ignContents)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	actualSource := *ignContents.Ignition.Config.Merge[0].Source
+	assert.EqualValues(t, expectedMasterIgnitionSource, actualSource, fmt.Sprintf("expected master pointer ignition to be %s but found %s", expectedMasterIgnitionSource, actualSource))
+}
+
+func verifyUpdateMCSCertKey(t *testing.T, bootstrap *bootstrap.Bootstrap) {
+	config := &igntypes.Config{}
+	config = bootstrap.Config
+
+	cert := &x509.Certificate{}
+	var data []byte
+
+	for i, fileData := range config.Storage.Files {
+		if fileData.Path == mcsCertFile {
+			contents := config.Storage.Files[i].Contents.Source
+			replaceable := "data:text/plain;charset=utf-8;base64,"
+			replaced := strings.Replace(*contents, replaceable, "", 1)
+
+			decodedCert, err := base64.StdEncoding.DecodeString(replaced)
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = yaml.Unmarshal(decodedCert, &data)
```

Comment:
I ran the unit tests against the code existing before your changes here, and it failed at this step.

Prior to your change, `decodedCert` here was just the certificate in plain text (starting with `-----BEGIN CERTIFICATE-----`).

After your change, `decodedCert` contains a base64-encoded string (so it's like it is being double-encoded). Strangely, the `yaml.Unmarshal` seems to have the effect of just undoing the base64 encoding (which I would not have predicted) and `data` ends up as the string starting with `-----BEGIN CERTIFICATE-----`).

If I set `data = decodedCert` with the fork code it passes, so I think the issue is just that you are double-encoding the certificate somewhere in this new code.